client/src/views/devices/GreetDeviceModal.vue

```diff
@@ -6,33 +6,48 @@
     :class="GreetDeviceStep[pageStep]"
   >
     <ms-wizard-stepper
-      v-show="pageStep > GreetDeviceStep.WaitForGuest && pageStep < GreetDeviceStep.Summary"
+      v-show="pageStep > GreetDeviceStep.WaitForGuest && pageStep < GreetDeviceStep.Summary && isLargeDisplay"
       :current-index="getStepperIndex()"
       :titles="['DevicesPage.greet.steps.hostCode', 'DevicesPage.greet.steps.guestCode']"
     />
     <ion-button
       slot="icon-only"
       @click="cancelModal()"
-      v-show="pageStep !== GreetDeviceStep.Summary"
       class="closeBtn"
+      v-if="pageStep !== GreetDeviceStep.Summary && isLargeDisplay"
     >
       <ion-icon
         :icon="close"
         class="closeBtn__icon"
       />
     </ion-button>
     <div class="modal">
-      <ion-header class="modal-header">
+      <ion-header
+        class="modal-header"
+        v-if="isLargeDisplay"
+      >
         <ion-title class="modal-header__title title-h2">
-          {{ $msTranslate(getTitleAndSubtitle().title) }}
+          {{ $msTranslate(getStep().title) }}
         </ion-title>
         <ion-text
           class="modal-header__text body"
-          v-show="getTitleAndSubtitle().subtitle"
+          v-show="getStep().subtitle"
         >
-          {{ $msTranslate(getTitleAndSubtitle().subtitle) }}
+          {{ $msTranslate(getStep().subtitle) }}
         </ion-text>
       </ion-header>
+      <small-display-step-modal-header
+        v-else
+        @close-clicked="cancelModal()"
+        :title="getStep().title"
+        :subtitle="getStep().subtitle"
+        :step="{
```

Comment:
I don't understand this prop.

It has a title but it's always the same, and we instead update the `title` prop of the component. It has a `total` which I suppose is the total number of step, but then it should be external to this prop.
![Screenshot 2025-05-16 061044](https://github.com/user-attachments/assets/9331f677-c019-4ba0-8e2b-e2daa2cea430)
It's backward, the title should be the top title, the step title should be linked to the current step.